### PR TITLE
Max validate

### PR DESCRIPTION
**Pull request by maxfroh**

### CODE REVIEW

File: src/Jott.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+public class Jott {
+    public static void main(String[] args) throws SyntaxException {
+        if (args.length == 0) {
+            System.out.println("Usage: java Jott <filename>");
+            return;
+        }
+
+        String filename = args[0];
+        ArrayList<Token> jottTokens = JottTokenizer.tokenize(filename);
+        JottTree jottTree = JottParser.parse(jottTokens);
```
Comment (liamlb663) [inline src/Jott.java:19]: shouldn't run the verification step and catch any errors.

File: src/group22/FunctionInfo.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package group22;
+
+public record FunctionInfo(DataType dataType) {
```
Comment (liamlb663) [inline src/group22/FunctionInfo.java:3]: This has since been implemented in scope manager I believe

File: src/group22/GrammarClasses/Else.java [hunk 3]
```diff
@@ -32,22 +39,32 @@ static Else parse(ArrayList<Token> tokens) throws SyntaxException {
...
         if (bodyNode == null) {
             return "";
         }
         return ("Else{" + bodyNode.convertToJott() + "}");
     }
 
-    public boolean validateTree() {
-        //TODO
-        return false;
+    public boolean validateTree(ScopeManager sm) {
```
Comment (liamlb663) [inline src/group22/GrammarClasses/Else.java:66]: I'm confused from the validateTree functions, none of these take args

File: src/Jott.java (hunk 1 above)
Comment (maxfroh) [reply src/Jott.java:19]: what do you mean

**Pull request closed without merging**